blocks/math/test/qa_Math_ext.cpp

```diff
@@ -0,0 +1,233 @@
+#include <cmath>
+
+/* ------------------------------------------------------------------ */
+/*                         shared test helpers                         */
+/* ------------------------------------------------------------------ */
+
+template<typename T>
+struct TestParameters {
+    std::vector<T>              input{};
+    std::vector<std::vector<T>> inputs{};
+    std::vector<T>              output{};
+};
+
+template<typename T, typename BlockUnderTest>
+void test_block(const TestParameters<T>& p)
+{
+    using namespace boost::ut;
+    using namespace gr;
+    using namespace gr::testing;          // TagSource / TagSink definitions
+    using namespace gr::blocks::math;
+
+    const Size_t n_inputs = static_cast<Size_t>(p.inputs.size());
+
+    Graph g;
+    auto& sink = g.emplaceBlock<TagSink<T, ProcessFunction::USE_PROCESS_ONE>>();
+
+    if (!p.input.empty()) {
+        /* single input */
+        auto& blk = g.emplaceBlock<BlockUnderTest>();
+        auto& src = g.emplaceBlock<TagSource<T>>(gr::property_map{
+                       {"values",        p.input},
+                       {"n_samples_max", static_cast<Size_t>(p.input.size())}});
+
+        expect(eq(g.connect(src, "out"s, blk, "in"s),
+                  ConnectionResult::SUCCESS));
+        expect(eq(g.connect<"out">(blk).template to<"in">(sink),
+                  ConnectionResult::SUCCESS));
+    } else {
+        /* multiple inputs */
+        auto& blk = g.emplaceBlock<BlockUnderTest>(
+                        gr::property_map{{"n_inputs", n_inputs}});
+
+        for (Size_t i = 0; i < n_inputs; ++i) {
+            auto& src = g.emplaceBlock<TagSource<T>>(gr::property_map{
+                           {"values",        p.inputs[i]},
+                           {"n_samples_max", static_cast<Size_t>(p.inputs[i].size())}});
+            expect(eq(g.connect(src, "out"s, blk, "in#"s + std::to_string(i)),
+                      ConnectionResult::SUCCESS));
+        }
+
+        expect(eq(g.connect<"out">(blk).template to<"in">(sink),
+                  ConnectionResult::SUCCESS));
+    }
+
+    scheduler::Simple sch{std::move(g)};
+    expect(sch.runAndWait().has_value());
+    expect(std::ranges::equal(sink._samples, p.output));
+}
+
+/* ------------------------------------------------------------------ */
+/*                                tests                                */
+/* ------------------------------------------------------------------ */
+
+const boost::ut::suite<"math-ext"> math_ext = [] {
+    using namespace boost::ut;
+    using namespace gr;
+    using namespace gr::blocks::math;
+
+    constexpr auto kArith = std::tuple<
+        uint8_t,uint16_t,uint32_t,uint64_t,
+        int8_t,int16_t,int32_t,int64_t,
+        float,double>();
+
+    constexpr auto kLogic = std::tuple<
+        uint8_t,uint16_t,uint32_t,uint64_t,
+        int8_t,int16_t,int32_t,int64_t>();
+
+    /* ---------- Max / Min ----------------------------------------- */
```

Comment:
you can embed the code in `"Max / Min subtests"_test = [] {...}"`. This would make this code more expressive and avoids the comments.